apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx

```diff
@@ -247,9 +274,108 @@ const GridHeaderActions = ({ table, isViewSelected, isTableSelected }: GridHeade
           )
         ) : null}
 
+        {/* need to check if view is security invoker or definer */}
+        {isView && viewIsSecurityDefiner && (
+          <Popover_Shadcn_ open={open} onOpenChange={() => setOpen(!open)} modal={false}>
+            <PopoverTrigger_Shadcn_ asChild>
+              <Button type="warning" icon={<Unlock strokeWidth={1.5} />}>
+                View is public
+              </Button>
+            </PopoverTrigger_Shadcn_>
+            <PopoverContent_Shadcn_ className="min-w-[395px] text-sm" align="end">
+              <h3 className="flex items-center gap-2">
+                <Unlock size={16} /> Secure your View
+              </h3>
+              <div className="grid gap-2 mt-4 text-foreground-light text-sm">
+                <p>
+                  Postgres' default setting for views is SECURITY DEFINER which means they use the
+                  permissions of the view's creator, rather than the permissions of the querying
+                  user when executing the view's underlying query.
+                </p>
+                <p>
+                  Creating a view in the public schema makes that view accessible via your project's
+                  APIs.
+                </p>
+
+                <div className="mt-2">
+                  <Button type="default" asChild>
+                    <Link
+                      target="_blank"
+                      href="https://supabase.com/docs/guides/database/database-advisors?lint=0010_security_definer_view"
+                    >
+                      Learn more
+                    </Link>
+                  </Button>
+                </div>
+              </div>
+            </PopoverContent_Shadcn_>
+          </Popover_Shadcn_>
+        )}
+
+        {isMaterializedView && entityType.schema === 'public' && (
+          <Popover_Shadcn_ open={open} onOpenChange={() => setOpen(!open)} modal={false}>
+            <PopoverTrigger_Shadcn_ asChild>
+              <Button type="warning" icon={<Unlock strokeWidth={1.5} />}>
+                View is public
+              </Button>
+            </PopoverTrigger_Shadcn_>
+            <PopoverContent_Shadcn_ className="min-w-[395px] text-sm" align="end">
+              <h3 className="flex items-center gap-2">
+                <Unlock size={16} /> Secure your View
+              </h3>
+              <div className="grid gap-2 mt-4 text-foreground-light text-sm">
+                <p>
+                  Materialized view is in the public schema and can be queried by anon and
```

Comment:
```suggestion
                  Materialized view in the public schema makes that view accessible via your project's
                  APIs.
```

To keep it consistent with the earlier message for a view